src/components/common/ButtonComponent.tsx

```diff
@@ -0,0 +1,23 @@
+import { MouseEventHandler } from "react";
+
+interface ButtonComponentProps {
+  className?: string;
+  onClick?: MouseEventHandler<HTMLButtonElement>;
+  children: React.ReactNode;
+  disabled: boolean;
+}
+const ButtonComponent = (props: ButtonComponentProps) => {
+  return (
+    <button
+      className={`${
+        props.className ? props.className : ""
+      } text-white text-center text-xs font-semibold bg-black self-center w-[185px] max-w-full justify-center items-center mt-6 px-16 py-4 max-md:px-5 disabled:opacity-30`}
```

Comment:
You can add a default value of the className to be `""`, then you avoid the conditional. 
Another thing, the way you have it here, the `className` received will be overwritten by the fixed classes you added. If you want it to be the other way round, the received className needs to be at the end.

Like this the component classes will override the received `className`
```suggestion

//at the definition of the component: const ButtonComponent = ({ className = "", onClick, children, disabled }: //ButtonComponentProps) => {

      className={`${
        props.className
      } text-white text-center text-xs font-semibold bg-black self-center w-[185px] max-w-full justify-center items-center mt-6 px-16 py-4 max-md:px-5 disabled:opacity-30`}
```


Like this the received classes will override:

```suggestion
className={`text-white text-center text-xs font-semibold bg-black self-center w-[185px] max-w-full justify-center items-center mt-6 px-16 py-4 max-md:px-5 disabled:opacity-30 ${props.className} `}
```